backend/backend.go

```diff
@@ -886,12 +886,16 @@ func (backend *Backend) CancelConnectKeystore() {
 	backend.connectKeystore.cancel(errUserAbort)
 }
 
-// SetWatchonly sets the app config's watchonly flag to `watchonly`.
-// When enabling watchonly, all currently loaded accounts are turned into watchonly accounts.
-// When disabling watchonly, all persisted accounts's watchonly status is reset.
-func (backend *Backend) SetWatchonly(watchonly bool) error {
-	err := backend.config.ModifyAppConfig(func(config *config.AppConfig) error {
-		config.Backend.Watchonly = watchonly
+// SetWatchonly sets the keystore's watchonly flag to `watchonly`.
+// When enabling watchonly, all currently loaded accounts of that keystore are turned into watchonly accounts.
+// When disabling watchonly, all the watchonly status of all of the keystore's persisted accounts is reset.
+func (backend *Backend) SetWatchonly(rootFingerprint []byte, watchonly bool) error {
```

Comment:
Later in this function, when you call `backend.AccountSetWatch` to set or reset the watch-only flag of the accounts, I think you should filter for keystore, otherwise the setting propagates to all the loaded accounts. This causes e.g. the removal of a watched device's accounts when disabling the watch-only for another device.

Reply:
Great  catch, thanks. Will fix!